erpnext/controllers/taxes_and_totals.py

```diff
@@ -643,7 +643,7 @@ def calculate_change_amount(self):
 				self.doc.base_write_off_amount, self.doc.precision("base_change_amount"))
 
 	def calculate_write_off_amount(self):
-		if flt(self.doc.change_amount) > 0:
+		if flt(self.doc.write_off_outstanding_amount_automatically) == 1:
```

Comment:
`test_pos_change_amount` is failing due to this. `write_off_outstanding_amount_automatically` needs to be enabled to calculate `write_off_amount`